docs/contributing/architecture/tests.md

```diff
@@ -1,24 +1,49 @@
 # Tests
 
-Tests in Cursorless are done in various ways depending on what is tested.
+Tests are critical to Cursorless. Our tests fall broadly into a three categories:
 
-1. **Cursorless tests for VSCode**: tests the actions (`take`, `chuck`, etc.) inside VSCode
+- **Recorded tests** are end-to-end / subcutaneous tests that run a full Cursorless command and check that the ide ends up in the correct state. These represent the bulk of our tests, and are generated automatically using the [test case recorder](../test-case-recorder.md)
+- **Scope tests** check that cursorless scopes are defined correctly for all languages that support the given scope. They consist of a small test document followed by `---`, and then a snapshot of the expected scopes in the given document. They are generated automatically using our scope test recorder, as described in [scope test recorder](../adding-a-new-scope.md#4-add-tests-for-the-given-scope)
+- **Other tests** include unit tests, handwritten subcutaneous / end-to-end tests, etc
 
-2. **Cursorless unit tests**: tests specific TypeScript functions (`toggle decorations`, `visible multiple regions`, etc.). These tests don't rely on VSCode or neovim
+We run the above tests in various contexts, both locally and in CI. The contexts are:
 
-3. **Cursorless tests for neovim**: tests the actions (`take`, `chuck`, etc.) inside neovim. 99% of them are the same as the "Cursorless tests for VSCode" except some new ones are needed in order to test all actions, due to no hats being supported yet in neovim (and the fact that lots of "Cursorless tests for VSCode" rely on hats)
+- **VSCode**: Today, many of our tests must run within a VSCode context. For some of our tests, this is desirable, because they are designed to test that our code works in VSCode. However, many of our tests (such as scope tests and recorded tests) are not really VSCode-specific, but we haven't yet built the machinery to run them in a more isolated context, which would be much faster.
```

Comment:
aren't the recorded tests in `recorded.vscode.test.ts` not VSCode-specific? I don't understand why you say they are not really VSCode-specific.

Reply:
Oh I see what you mean, you talk about the fixtures.

Reply:
Running them in an isolated context would not remove the need to run them in the vscode context, right? Might be worth clarifying your sentence as it seems to imply we would not need to run it in vscode anymore?

Reply:
yeah we eventually may stop running them in vscode as it is very slow